### PR TITLE
DC-940: Cohort Builder - Select cohort by default

### PR DESCRIPTION
**Pull request by snf2ye**
## Summary of changes:
- On Create of cohort, we automatically add it to the list of selected cohorts

## Why:
- Improvement to the user experience: We're now assuming that if you create a cohort, you want to include it in the snapshot access request

## Testing
- added check on unit test

https://github.com/user-attachments/assets/000fc98d-3743-4dcb-b554-f12029ffae71

### CODE REVIEW

File: src/dataset-builder/CohortEditor.ts [hunk 1]
```diff
@@ -573,12 +573,20 @@ interface CohortEditorProps {
   readonly snapshotBuilderSettings: SnapshotBuilderSettings;
   readonly originalCohort: Cohort;
   readonly updateCohorts: Updater<Cohort[]>;
+  readonly updateSelectedCohortsOnCreate: (cohort: Cohort) => void;
```
Comment (s-rubenstein) [inline src/dataset-builder/CohortEditor.ts]: Nit: How would you feel about just calling this `updateSelectedCohorts`? I am a bit concerned that `OnCreate` is a detail about the usage not about the prop itself. (Or maybe `addSelectedCohort`, actually? Since its only a single cohort?)

File: src/dataset-builder/DatasetBuilder.ts [hunk 2]
```diff
@@ -669,7 +659,10 @@ export const DatasetBuilderView: React.FC<DatasetBuilderProps> = (props) => {
     initialState || homepageState.new()
   );
   const [cohorts, setCohorts] = useState<Cohort[]>([]);
-  const [selectedCohorts, setSelectedCohorts] = useState([] as HeaderAndValues<Cohort>[]);
+  const defaultHeader = 'Saved cohorts';
```
Comment (s-rubenstein) [inline src/dataset-builder/DatasetBuilder.ts]: What do you think about moving this up so it isn't mixed in with the state initialization?

**Review by s-rubenstein — COMMENTED** [review]
Looks pretty reasonable to me, pending tests

File: src/dataset-builder/DatasetBuilder.ts [hunk 3]
```diff
@@ -324,7 +311,7 @@ export const CohortSelector = ({
       objectSets: [
         {
           values: cohorts,
-          header: 'Saved cohorts',
+          header: cohortGroupName,
```
Comment (rjohanek) [inline src/dataset-builder/DatasetBuilder.ts]: why the change to the header? Will it be different from the default value at times?

File: src/dataset-builder/CohortEditor.ts (hunk 1 above)
Comment (snf2ye) [reply src/dataset-builder/CohortEditor.ts]: Let me know what you think of the new names - they ended up getting reworked when I made the utils method more generic.

**Review by rjohanek — APPROVED** [review]
looks good thanks. one question.

File: src/dataset-builder/DatasetBuilder.ts [hunk 4]
```diff
@@ -682,6 +671,10 @@ export const DatasetBuilderView: React.FC<DatasetBuilderProps> = (props) => {
     return criteriaCount;
   };
 
+  const addSelectedCohort = (cohort: Cohort) => {
+    addSelectableObjectToGroup(cohort, defaultHeader, selectedCohorts, setSelectedCohorts);
```
Comment (rjohanek) [inline src/dataset-builder/DatasetBuilder.ts:675]: why are these called add and not just update?

Comment (snf2ye) [reply src/dataset-builder/DatasetBuilder.ts:675]: "add" is a little more precise language around the operation occurring here: We're adding a single cohort to a list of cohorts. But, if "update" is more in line with language used around Terra UI, that would also be fine.

File: src/dataset-builder/CohortEditor.ts [hunk 5]
```diff
@@ -610,6 +618,10 @@ export const CohortEditor: React.FC<CohortEditorProps> = (props) => {
             onClick: () => {
               updateCohorts((cohorts) => {
                 const index = _.findIndex((c) => _.equals(c.name, cohort.name), cohorts);
+                if (index === -1) {
+                  // Only add to selectedCohorts on creation of new cohort
+                  addSelectedCohort(cohort);
```
Comment (pshapiro4broad) [inline src/dataset-builder/CohortEditor.ts:623]: Did you consider doing this inside `updateCohorts` instead? If the rule is we want to select any cohort that was added, this method could check the set of cohorts before and after an update and then add the new cohorts to the selection. That would avoid the need to pass down two different methods to a subcomponent, since the code that manages the selection also manages the cohort list state.

File: src/dataset-builder/DatasetBuilder.ts [hunk 6]
```diff
@@ -324,7 +309,7 @@ export const CohortSelector = ({
       objectSets: [
         {
           values: cohorts,
-          header: 'Saved cohorts',
+          header: selectedCohorts?.length > 0 ? selectedCohorts[0].header : '',
```
Comment (pshapiro4broad) [inline src/dataset-builder/DatasetBuilder.ts]: I'm wondering about this part of the change. Does this display need a header at all? What is it telling the user? I think it needs a prompt when the list is empty, but I don't think it needs a header if there's a cohort present.

Comment (snf2ye) [reply src/dataset-builder/DatasetBuilder.ts]: I don't think this needs a header in order to display it. But, the selector for this object set is used in other spots where it does need a header. And I think we use the header to group the cohorts (So, I don't think it matters what it is, as long as it is the same across cohorts). 
I think this would be a bigger refactor to move away from setting the header, but I'm new to this codebase so maybe it is not!

File: src/dataset-builder/CohortEditor.ts (hunk 5 above)
Comment (snf2ye) [reply src/dataset-builder/CohortEditor.ts:623]: UpdateCohorts is directly calling the setState for cohorts and is used rather broadly. We only pass addSelectedCohort() into one child method. 

Is there a straightforward way to compare the arrays and only get the new items, maybe with lodash? I'd be worried that logic would be complicated/hard to read.

**Review by pshapiro4broad — APPROVED** [review]

File: src/dataset-builder/DatasetBuilder.ts (hunk 6 above)
Comment (pshapiro4broad) [reply src/dataset-builder/DatasetBuilder.ts]: For the record, the header is needed because the component that displays cohorts has a header (even though all cohorts have the same header), and the selection code uses the header to find the object to select. This is required in the case where the same name is in different header groups (with different headers) so the selection code knows how to find it.
